### ISSUE TITLE
Unexpected inferred type parsing partial property doc comment

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#291]
Observed: `Argument 1 of fopen expects string, $fooPropTypo provided`
Expected: No error

```php
<?php
class MyClass {
    /**
     * Comment
     * @var $fooPropTypo string
     */
    public $fooProp = '/tmp/file.txt';

    /** @return resource|false */
    public function open() {
        $result = fopen($this->fooProp, 'r');
        return $result;
    }
}
```

**Comment by muglug** [comment]
Should have a `InvalidDocblock` issue there

**Comment by TysonAndre** [comment]
I'd disagree on the `InvalidDocblock`, partly because the spec is vague to me. I don't see "**MUST** have a type" in the documentation, and `["Type"]` is in square brackets, like the description.

- http://docs.phpdoc.org/references/phpdoc/tags/var.html
- > @var [“Type”] [$element_name] [<description>]

I sometimes write `/** @var $x description of what x is used for */`

**Comment by muglug** [comment]
How about `MissingVarDocblockType` if the first thing after the `@var` begins with `$` and isn't `$this`?

**Comment by TysonAndre** [comment]
MissingVarDocblockType is fine with me

**Comment by muglug** [comment]
Docblocks are conquered... for now.
